checker/src/org/checkerframework/checker/lock/LockVisitor.java

```diff
@@ -4,88 +4,325 @@
+                    if (AnnotationUtils.hasElementValue(annotationMirror, "value")) {
+                        List<String> guardedByValue = AnnotationUtils.getElementValueArray(annotationMirror, "value", String.class, false);
+
+                        for(String lockExpression : guardedByValue) {
+                            if (translateItselfToThis && lockExpression.equals("itself")) {
+                                lockExpression = "this";
+                            }
+                            preconditions.add(Pair.of(lockExpression, LockHeld.class.toString().substring(10 /* "interface " */)));
+                        }
+                    }
+                }
+            }
+        }
+
+        return preconditions;
+    }
+
+    @Override
+    protected void checkAccess(IdentifierTree node, Void p) {
+        AnnotatedTypeMirror atm = atypeFactory.getAnnotatedType(node);
+        boolean hasJavaxOrJcipGuardedByAnnotation = atm == null ? false : (atm.hasAnnotation(javaxGuardedByClass) || atm.hasAnnotation(jcipGuardedByClass));
+        // Don't check preconditions for annotations on javax or JCIP @GuardedBy because
+        // those are meant for uses of those annotations when they are acting as the @Holding annotation,
+        // not when they are acting as the Lock Checker's @GuardedBy type qualifier.
+        checkAccess(node, p, !hasJavaxOrJcipGuardedByAnnotation);
     }
 
     @Override
     protected void commonAssignmentCheck(AnnotatedTypeMirror varType,
             AnnotatedTypeMirror valueType, Tree valueTree, /*@CompilerMessageKey*/ String errorKey,
-            boolean isLocalVariableAssignement) {
+            boolean isLocalVariableAssignment) {
+
+        Kind valueTreeKind = valueTree.getKind();
+
+        switch(valueTreeKind) {
+            case NEW_CLASS:
+            case NEW_ARRAY:
+                // Avoid issuing warnings for: @GuardedBy(<something>) Object o = new Object();
+                // Do NOT do this if the LHS is @GuardedByBottom.
+                if (!varType.hasAnnotation(GuardedByBottom.class))
+                    return;
+                break;
+            case INT_LITERAL:
+            case LONG_LITERAL:
+            case FLOAT_LITERAL:
+            case DOUBLE_LITERAL:
+            case BOOLEAN_LITERAL:
+            case CHAR_LITERAL:
+            case STRING_LITERAL:
+                // Avoid issuing warnings for: @GuardedBy(<something>) Object o; o = <some literal>;
+                // Do NOT do this if the LHS is @GuardedByBottom.
+                if (!varType.hasAnnotation(GuardedByBottom.class))
+                    return;
+                break;
+            case NULL_LITERAL: // Assigning null to any LHS is valid.
+                return;
+            default:
+        }
+
+        boolean skipSubtypeCheck = false;
+
+        // Assigning a value with a @GuardedBy annotation to a variable with a @GuardSatisfied annotation is always
```

Comment:
This is still needed. It is documented in the manual in the section about polymorphic assignments.